src/ucp/core/ucp_ep.c

```diff
@@ -1307,6 +1352,16 @@ static void ucp_ep_discard_lanes(ucp_ep_h ep, ucs_status_t discard_status)
         return;
     }
 
+    failed_arg = ucs_malloc(sizeof(*failed_arg), "failed_arg");
+    if (failed_arg == NULL) {
+        failed_arg = &ucp_failed_tl_ep;
+    } else {
```

Comment:
why need to handle it this way? if this memory allocation fails, then memory allocation below will probably fail too.
wdyt about just doing the same actions as we have when handling failed memory allocation for `discard_arg` below?